src/strategies/DSL/actions/limitSwapAction.ts

```diff
@@ -1,17 +1,79 @@
-import { LimitSwapActionArgs, PrimitiveArgs } from '@brinkninja/types'
+import { toTokenArgs } from '../../../internal/toTokenArgs'
+import { LimitSwapActionArgs, PrimitiveArgs, PrimitiveParamValue } from '@brinkninja/types'
+import priceCurveAddress from '../../../internal/priceCurveAddress'
+import { convertToHexPrice } from '@brink-sdk/internal/toHexPrice';
+
+const DEFAULT_PRICE_CURVE = 'flat';
+
+const DECIMALS = 18;
+const FACTOR = BigInt(10 ** DECIMALS);
+
+let globalPriceCurveAddress: string | null = null;
+let addressFetchInitiated = false;
+
+function ensurePriceCurveAddressInitialized(): void {
```

Comment:
If someone knows how to better handle it, i'm interested. using async functions inside sync functions does not work well.

i wonder if we should just make everything async? 😆

Reply:
ok. what about we just const FLAT_PRICE_CURVE?